packages/lending/src/protocols/aave-v2/lending-protocol.ts

```diff
@@ -329,34 +328,32 @@ export class LendingProtocol extends Protocol {
     return isAToken(this.chainId, token);
   }
 
-  async newSupplyLogic(params: SupplyParams) {
-    const supplyQuotation = await apisdk.protocols.aavev2.getDepositQuotation(this.chainId, {
-      input: params.input,
-      tokenOut: toAToken(this.chainId, params.input.token),
+  async newSupplyLogic({ marketId, input }: TokenInFields) {
+    return apisdk.protocols.aavev2.newDepositLogic({
+      input,
+      output: new common.TokenAmount(this.toProtocolToken(marketId, input.token), input.amount),
     });
-    return apisdk.protocols.aavev2.newDepositLogic({ ...supplyQuotation, balanceBps: common.BPS_BASE });
   }
 
-  async newWithdrawLogic(params: WithdrawParams) {
-    const withdrawQuotation = await apisdk.protocols.aavev2.getWithdrawQuotation(this.chainId, {
-      input: {
-        token: toAToken(this.chainId, params.output.token),
-        amount: params.output.amount,
-      },
-      tokenOut: params.output.token,
+  async newWithdrawLogic({ marketId, output }: TokenOutFields) {
+    return apisdk.protocols.aavev2.newWithdrawLogic({
+      input: new common.TokenAmount(this.toProtocolToken(marketId, output.token), output.amount),
+      output,
     });
-    return apisdk.protocols.aavev2.newWithdrawLogic(withdrawQuotation);
   }
 
-  newBorrowLogic = apisdk.protocols.aavev2.newBorrowLogic;
+  newBorrowLogic({ output }: TokenOutFields) {
+    return apisdk.protocols.aavev2.newBorrowLogic({
+      output,
+      interestRateMode: logics.aavev2.InterestRateMode.variable,
```

Comment:
I think it's nice to use the `interestRateMode` from `logics.aavev2`. However, the `defaultInterestRateMode` in `adapter.ts` should be adjusted as well.